src/firebase_functions/pubsub.py

```diff
@@ -32,49 +30,87 @@ class CloudEvent(Generic[T]):
 class Message(Generic[T]):
   message_id: str
   publish_time: dt.datetime
-  data: str
-  attributes: dict[str, str]
-  ordering_key: str
+  data: Optional[str] = None
+  attributes: Optional[dict[str, str]] = None
+  ordering_key: Optional[str] = None
 
   @property
-  def json(self) -> Dict[str, Any]:
-    return {
+  def json(self) -> Optional[T]:
```

Comment:
I wonder if this should just be:
```suggestion
  def json(self) -> T:
```

Since user can specify optionality, e.g. if they say `T `is `Optional[dict]` we don't want it becoming `Optional[Optional[dict]]`?

Reply:
I made it Optional for this:
![Screen Shot 2022-08-08 at 5 52 25 PM](https://user-images.githubusercontent.com/41123719/183447090-bc7a5ad8-f882-4881-9959-191c18991805.png)

Cause data might be None

Reply:
Another approach tho:
```python
  @property
  def json(self) -> T:
    try:
      data = self.data or ''
      return json.loads(base64.b64decode(data).decode('utf-8'))
```

Reply:
we'll ignore this for now then, thanks